test/e2e/gateway/gateway_test.go

```diff
@@ -17,42 +17,95 @@ package gateway
 
 import (
 	"context"
+	"fmt"
 	"testing"
 
 	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gexec"
+	"github.com/projectcontour/contour/pkg/config"
 	"github.com/projectcontour/contour/test/e2e"
 	"github.com/stretchr/testify/require"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	gatewayv1alpha1 "sigs.k8s.io/gateway-api/apis/v1alpha1"
 )
 
+var f = e2e.NewFramework(false)
+
 func TestGatewayAPI(t *testing.T) {
+	RegisterFailHandler(Fail)
 	RunSpecs(t, "Gateway API tests")
 }
 
+var _ = BeforeSuite(func() {
+	require.NoError(f.T(), f.Deployment.EnsureResourcesForLocalContour())
+})
+
+var _ = AfterSuite(func() {
+	f.DeleteNamespace(f.Deployment.Namespace.Name, true)
+	gexec.CleanupBuildArtifacts()
+})
+
 var _ = Describe("Gateway API", func() {
-	var f *e2e.Framework
+	var (
+		contourCmd            *gexec.Session
+		contourConfig         *config.Parameters
+		contourConfigFile     string
+		additionalContourArgs []string
+	)
+
+	// Creates specified gateway in namespace and runs namespaced test
+	// body. Modifies contour config to point to gateway.
+	testWithGateway := func(gw *gatewayv1alpha1.Gateway, body e2e.NamespacedTestBody) e2e.NamespacedTestBody {
```

Comment:
curious what people think of the patterns in this test

Reply:
:+1: this looks nice